app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,87 @@
+class TweetsController < ApplicationController
+
+
+	get '/tweets' do
+		if logged_in?
+			@user = current_user
+			success
+			erb :'/tweets/tweets'
+		else
+			please_login
+			redirect '/login'
+		end
+	end
+
+
+	get '/tweets/new' do
+		if logged_in?
+			@user = current_user
+			erb :'/tweets/create_tweet'
+		else
+			please_login
+			redirect '/login'
+		end
+	end
+
+	post '/tweets/new' do
+		if filled_out(params)
+			@tweet = Tweet.create(content: params[:content])
+			@tweet.id = Tweet.all.last.id
+			@tweet.user_id = current_user.id
+			@tweet.save
+			success
+			redirect '/tweets'
+		else
+			not_filled_out
+			redirect '/tweets/new'
+		end
+	end
+
+
+	get '/tweets/:id' do
+		if logged_in?
+			@tweet = Tweet.find(params[:id])
+			erb :'/tweets/show_tweet'
+		else
+			please_login
+			redirect '/login'
+		end
+	end
+
+
+	get '/tweets/:id/edit' do
+		if logged_in?
+			@tweet = Tweet.find(params[:id])
+			erb :'/tweets/edit_tweet'
+		else
+			please_login
+			redirect '/login'
+		end
+	end
+
+	post '/tweets/:id/edit' do
+		@tweet = Tweet.find(params[:id])
+			if current_user.id == @tweet.user_id
+				@tweet.update(content: params[:content])
+				success
+				erb :'/tweets/show_tweet'
+			else
+				failure
+				redirect "/tweets/#{@tweet.id}"
+			end
+	end
+
+
+	delete '/tweets/:id/delete' do
+		@tweet = Tweet.find(params[:id])
+			if current_user.id == @tweet.user_id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.